### ISSUE TITLE
[Bug]: failing silently when branch is master, not main

### ISSUE THREAD

**Issue by kurko** [ObsidianPublisher/obsidian-github-publisher#33]
### Issue validation

- [X] My issue is not a question relative to [Material Mkdocs](https://squidfunk.github.io/mkdocs-material/) nor [Mkdocs](https://www.mkdocs.org/) (*[You can open a discussion to get more help!](https://github.com/squidfunk/mkdocs-material/discussions)*)
- [X] I checked the issue to prevent duplicate
- [X] I checked my configurations files and the documentation

### Workflow

Obsidian plugin 💠

### Describe the bug

tl;dr the plugin expects the repository's branch `main` to exist and fails silently when it's `master`.

It took me a good chunk of time to figure out why errors were happening. The popup had a generic error message that didn't help. Upon investigation I found out that when creating a PR, it requires a base branch `main`, which in my case didn't exist.

<img width="455" alt="Screen Shot 2022-10-08 at 22 32 13" src="https://user-images.githubusercontent.com/70312/194736026-99343d20-6f63-47d5-bd57-9e6a0b30c584.png">

Two problems:

1. the main one is that it fails silently. I think we should show a more explicit error on the screen.
2. it's not uncommon for someone to be using `master`. [This repo uses master](https://github.com/ObsidianPublisher/obsidian-github-publisher/branches) and is an example.

I think we could:

1. raise an error message that is explicit so the user knows what's going on.
2. add to the README the fact that we only support `main` (or add a `settings` entry to select a branch).

### How to reproduce ?

1. Create a repo without branch `main`.
1. Try to send files to that repo

### Configuration

```json
{
  "githubRepo": "blog",
  "githubName": "kurko",
  "GhToken": "ghp_[REDACTED]",
  "shareKey": "share",
  "ExcludedFolder": "",
  "fileMenu": true,
  "editorMenu": true,
  "downloadedFolder": "fixed",
  "folderDefaultName": "obsidian",
  "yamlFolderKey": "",
  "rootFolder": "",
  "workflowName": "",
  "embedImage": true,
  "defaultImageFolder": "images/obsidian",
  "autoCleanUp": true,
  "autoCleanUpExcluded": "",
  "folderNote": false,
  "convertWikiLinks": true,
  "convertForGithub": true,
  "subFolder": "",
  "embedNotes": false,
  "copyLink": false,
  "mainLink": "",
  "linkRemover": "",
  "hardBreak": false,
  "logNotice": false,
  "convertDataview": true,
  "useFrontmatterTitle": true,
  "censorText": [
    {
      "entry": "\\]\\(\\.\\/",
      "replace": "]({% link obsidian/"
    },
    {
      "entry": "\\({% link(.*)\\)",
      "replace": "({% link $1.md %})"
    }
  ],
  "inlineTags": false,
  "dataviewFields": [],
  "excludeDataviewValue": [],
  "metadataFileFields": [
    "obsidian"
  ],
  "frontmatterTitleKey": "filename",
  "shareExternalModified": false
}
```

### OS

MacOS

**Comment by Lisandra-dev** [comment]
Good catch ! I was pretty sure that I prevents this. Instead I will add a main branch settings. It Will be better :)

**Comment by Lisandra-dev** [comment]
I fixed it in the last beta 4.3.7-0.
